File: Latihan b/php/Memory.php
<?php
// deklarasi kelas memory inherit hardware
class Memory extends Hardware{
	//variabel lokal
	private $frequency;
	private $memorysize;
	private $supportscuda;
	//konstruktor
	public function __construct(){
		$this->frequency = 0;
        $this->memorysize = 0;
        $this->supportscuda = 0;
	}
	/*
	public function __construct($frequency = 0, $memorysize = 0, $supportscuda = 0){
		$this->frequency = $frequency;
		$this->memorysize = $memorysize;
		$this->supportscuda = $supportscuda;
	}
	*/
	//get dan set setiap variabel
	public function setfrequency($frequency){
		$this->frequency = $frequency;
	}
	public function getfrequency(){
		return $this->frequency;
	}
	public function setmemorysize($memorysize){
		$this->memorysize = $memorysize;
	}
	public function getmemorysize(){
		return $this->memorysize;
	}
	public function setsupportscuda($supportscuda){
		$this->supportscuda = $supportscuda;
	}
	public function getsupportscuda(){
		return $this->supportscuda;
	}
	public function __destruct(){
	}
}
?>

File: Latihan a/java/memory.java
// deklarasi kelas memori inherit hardware
class memory extends hardware{
    // variabel privat
    private int frequency;
    private int memorysize;
    private int supportcuda;
    // method public dan konstruktor
    public memory(){

    }
    public memory(int frequency, int memorysize, int supportcuda){
        this.frequency = frequency;
        this.memorysize = memorysize;
        this.supportcuda = supportcuda;
    }
    // get dans set semua variabel
    public int getfrequency(){
        return frequency;
    }
    public void setfrequency(int frequency){
        this.frequency = frequency;
    }
    public int getmemorysize(){
        return memorysize;
    }
    public void setmemorysize(int memorysize){
        this.memorysize = memorysize;
    }
    public int getsupportcuda(){
        return supportcuda;
    }
    public void setsupportcuda(int supportcuda){
        this.supportcuda = supportcuda;
    }
}